File: LeetCode/[M] - 1161. Maximum Level Sum of a Binary Tree/solution.go
package main

// TreeNode - Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxLevelSum(root *TreeNode) int {
	queue := make([]*TreeNode, 0)
	queue = append(queue, root)
	maxSum := root.Val
	level, currLevel := 1, 0
	for len(queue) != 0 {
		currLevel++
		currSum := 0
		lenQ := len(queue)
		for i := 0; i < lenQ; i++ {
			v := queue[0]
			queue = queue[1:]
			currSum += v.Val
			if v.Left != nil {
				queue = append(queue, v.Left)
			}
			if v.Right != nil {
				queue = append(queue, v.Right)
			}
		}
		if currSum > maxSum {
			maxSum = currSum
			level = currLevel
		}
	}
	return level
}

File: LeetCode/[M] - 1161. Maximum Level Sum of a Binary Tree/solution.py
import collections
from typing import Optional


# TreeNode - Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        queue = collections.deque([root])
        lenQ = 1
        maxSum = root.val
        level, currLevel = 1, 0
        while queue:
            currLevel += 1
            currSum = 0
            for _ in range(lenQ):
                v = queue.popleft()
                currSum += v.val
                lenQ -= 1
                if v.left:
                    queue.append(v.left)
                    lenQ += 1
                if v.right:
                    queue.append(v.right)
                    lenQ += 1
            if currSum > maxSum:
                maxSum = currSum
                level = currLevel
        return level
